File: 0198-house-robber/0198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            if i == 1:
                nums[i] = max(nums[0],nums[1])
            if i > 1:
                nums[i] = max(nums[i - 1], nums[i] + nums[i - 2])
        return nums[-1]

File: 0198-house-robber/0198-house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {

        for(int i = 1; i < nums.size(); i++){
            if(i == 1){
                nums[i] = max(nums[i-1], nums[i]);
            }
            else{
                nums[i] = max(nums[i-1], nums[i] + nums[i-2]);
            }
            
        }
        return nums[nums.size() - 1];
    }
};